lib/pipeline.js

```diff
@@ -1544,6 +1573,26 @@ class PipelineModel extends BaseModel {
                 })
             );
 
+            // Get downtime events
+            const downtimeEvents = metrics.filter(m => m && m.isDowntimeEvent);
+
+            if (downtimeEvents.length) {
+                let i = 0;
+
+                // Add downtime duration info
+                for (; i < downtimeEvents.length - 1; i += 1) {
+                    downtimeEvents[i].downtimeDuration = dayjs(downtimeEvents[i + 1].maxEndTime).diff(
+                        dayjs(downtimeEvents[i].maxEndTime),
+                        'second'
+                    );
+                }
+
+                downtimeEvents[i].downtimeDuration = dayjs(new Date()).diff(
+                    dayjs(downtimeEvents[i].maxEndTime),
+                    'second'
+                );
+            }
```

Comment:
Probably use map fn?

Reply:
need to do something different with last element in array
gets the duration based off current time vs next element's time